File: Arcade/Intro/javascript/differentSquares.js
function differentSquares(matrix) {
    let set = [];
    
    for(let x = 0; x < matrix.length - 1; x++) {
        for(let y = 0; y < matrix[0].length - 1; y++) {
            let aux = "";
            for(let i = x; i < x + 2; i++) {
                for(let j = y; j < y + 2; j++) aux += matrix[i][j];
            }
            if(!set.includes(aux)) set.push(aux);
        }
    }
    
    return set.length
}

File: Arcade/Intro/Ruby/differentSquares.rb
def differentSquares(matrix)
    set = []
    (matrix.size - 1).times do |x|
        (matrix[0].size - 1).times do |y|
            aux = ""
            (x..x+1).to_a.each{|i|
            (y..y+1).to_a.each{|j| aux += matrix[i][j].to_s}}
            set.push aux if !set.include?(aux)
        end
    end
    return set.size
end